dask_searchcv/adaptive.py

```diff
@@ -0,0 +1,447 @@
+            logger.info("New best val_score={} found".format(best_['val_score']))
+
+        if len(models) == 1:
+            break
+
+    times = toolz.groupby('id', times)
+    times = {k: [toolz.keyfilter(lambda key: key != 'id', vi) for vi in v]
+             for k, v in times.items()}
+    times = {k: toolz.merge_with(sum, v) for k, v in times.items()}
+    return {'history': history, 'final_val_scores': final_val_scores,
+            'params': params, 'models': models, 'times': times}
+
+
+class Hyperband(DaskBaseSearchCV):
+    """
+    An adaptive model selection method that trains models with
+    ``model.partial_fit`` and evaluates models with ``model.score``.
+
+    Attributes
+    ----------
+    cv_results_ : list of dicts
+        This is a list of all models tuned, and information about those
+        models.  This list of dicts can be imported into Pandas with
+        ``pd.DataFrame(self.cv_results_)``. Each dictionary contains the
+        keys ``test_score``, ``params`` and ``param_{p}`` for each ``p``
+        in params. That is, ``params['alpha'] == params_alpha``.
+    best_params_ : dict
+        The best parameters found by this model selection algorithm. This
+        is determined by cross validation.
+    best_estimator_ : type(model)
+        The best estimator as determined by this model selection algorithm.
+        This is determined by cross validation.
+    best_index_ : int
+        The index of the best performing estimator and parameters in
+        ``best_estimator_`` and ``best_params_``
+    history : list of dicts
+        The recorded history for this adaptive process. It includes the
+        evaluations on each step, and it's possible to use this to see each
+        decision on which models to keep evaluating. Each dictionary in this
+        variables has ``set(keys) == {}``
+
+    Methods
+    -------
+    fit(X, y, dry_run=False, **fit_kwargs)
+        Find the best classifier.
+    info()
+        Get info about finding the best classifier (i.e., how many calls to
+        ``partial_fit``?).
+
+
+    Notes
+    -----
+    This implements the `Hyperband model selection algorithm`_
+    and only requires one input, ``max_iter`` which is porprotional to
+    "computational effort" required. Notably it does not require balancing
+    a trade-off between the number of params to evaluate and how long to
+    train each config.
+
+    Hyperband has some theoritical guarantees on finding (close to) the
+    optimal config from ``params`` given this computational effort
+    ``max_iter``.
+
+    References
+    ----------
+    - Hyperband: A Novel Bandit-Based Approach to Hyperparameter Optimization, 2016.
+      Lisha Li, Kevin Jamieson, Giulia DeSalvo, Afshin Rostamizadeh, Ameet Talwalkar
+      https://arxiv.org/abs/1603.06560
+
+    """
+    def __init__(self, model, params, max_iter=81, eta=3, n_jobs=-1, **kwargs):
+        """
+        Parameters
+        ----------
+        params :
```

Comment:
This should be named `param_grid` to match sklearn.

Reply:
Also, flip the order in the docs (the function signature is right).

Reply:
I chose to mirror the naming of RandomizedSearchCV.